grammar.js

```diff
@@ -263,7 +266,13 @@ module.exports = grammar({
       field('name', choice($.constant, $.scope_resolution)),
       field('superclass', optional($.superclass)),
       $._terminator,
-      $._body_statement
+      optional(field("namespace_body", $.body)),
+      'end'
+    ),
+
+    body: $ => choice(
+      seq($._statements, repeat(choice($.rescue, $.else, $.ensure))),
```

Comment:
This is a duplicated and slightly modified version of `_body_statement`.  At first I wondered why you didn't just define `body: $ => $._body_statement`. I think this is because you quite rightly did not want to have the `end` keyword be part of the body. 


What about writing the rules as follows
```
_body_statement: $ => choice(
  seq($._statements, repeat(choice($.rescue, $.else, $.ensure))),
  seq(optional($._statements), repeat1(choice($.rescue, $.else, $.ensure))),
),

body_statement: $ => $._body_statement,

begin: $ => seq('begin', optional($._terminator), optional($._body_statement), 'end'),
```